### PR TITLE
Fix #71 by toggling the start button and showing a warning for invali…

### LINKED ISSUE #71: Submitting a random string on the home page select LAD box returns a blank map
Whilst looking through the log data I saw a couple of entries that didn't match available LADs and Transport authorities on the app.

It looks like some users might type in their authority and hit `Start` without selecting from the presented drop-down.

E.g. https://atip.uk/scheme.html?authority=narnia#1/0/0 

Screenshot 2023-02-08 at 11 32 21

## Steps to reproduce

1. Visit https://atip.uk/index.html
2. In the `Select Transport Authority or Local Authority District:` box enter your own random string
3. Click Start

## Suggested fix

I think we'd want to see if we could write in some user validation at the select step, just to make sure they are constrained to available LADs/authorities.

### PR DESCRIPTION
**Pull request by dabreegster**
…d input

https://user-images.githubusercontent.com/1664407/218427282-29e56454-0e7e-406d-9541-189eb49dd678.mp4

### CODE REVIEW

**Comment by dabreegster** [comment]
This is an alternative to #74 that constantly updates the button state and warning text upon every keypress, not just when you click the start button. Any thoughts?

Putting the warning next to the button felt a bit more clear to me than a new paragraph, but not really sure where it belongs

File: ChooseArea.svelte [hunk 1]
```diff
@@ -9,8 +9,11 @@
 
   let showAbout = false;
 
-  let inputList;
+  let inputValue;
   let dataList;
+  let authoritySet = new Set();
+
+  $: validEntry = authoritySet.has(inputValue);
```
Comment (dabreegster) [inline ChooseArea.svelte:16]: https://svelte.dev/tutorial/reactive-declarations makes it trivial for us to recalculate this when the input changes

File: ChooseArea.svelte [hunk 2]
```diff
@@ -103,9 +105,16 @@
   <br />
 
   <p>Select Transport Authority or Local Authority District:</p>
-  <input list="authorities-list" bind:this={inputList} />
-  <datalist id="authorities-list" bind:this={dataList} />
-  <button type="button" on:click={start}>Start</button>
+  <div>
+    <input list="authorities-list" bind:value={inputValue} />
```
Comment (dabreegster) [inline ChooseArea.svelte:109]: I realized it was easier to just `bind:value` instead of binding a DOM element and then doing `x.value`

**Comment by Robinlovelace** [comment]
Taking a look now...

**Comment by Robinlovelace** [comment]
Looking at the video you posted Dustin and trying what's currently on atip.uk, I approve of this, seems like a good solution.

**Review by Robinlovelace — APPROVED** [review]

**Comment by Sparrow0hawk** [comment]
Much sveltier! 

My only comment would be should we just drop the warning text and have the button disabled until someone picks a valid option from the list?

But that's just a personal UX take.

**Comment by Robinlovelace** [comment]
Nice adjective: svelty!

**Comment by Robinlovelace** [comment]
Nice adjective: svelty!

**Comment by dabreegster** [comment]
Interpreting closure of #74 as being in favor of this, so merging

**Comment by dabreegster** [comment]
Oops, wifi here is weird and didn't pick up new messages, but let me submit?

> My only comment would be should we just drop the warning text and have the button disabled until someone picks a valid option from the list?

Agreed... it's clear enough that the button is disabled. Done in e7f70ba37007cd5fb14e2526b0f90eb2618036f8

**Pull request merged**